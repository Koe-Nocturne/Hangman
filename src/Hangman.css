.Hangman {
  width: 200px;
  text-align: center;
}

.Hangman-word {
  letter-spacing: 1em;
}
