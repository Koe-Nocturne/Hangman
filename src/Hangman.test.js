import React from "react";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
import Hangman from "./Hangman";

it("renders without crashing", function () {
  shallow(<Hangman />);
});

it("matches snapshot after click", function () {
  let wrapper = shallow(<Hangman />);

  wrapper
    .find("button[value='z']")
    .simulate("click", { target: { value: "z" } });

  let img_elem = wrapper.find("img").first().html();
  let serialized = toJson(wrapper);
  
  expect(img_elem).toContain('src="1.jpg"');
  expect(serialized).toMatchSnapshot();

});

it("image doesn't disapears after 6 wrong guesses", function(){
  let wrapper = mount(<Hangman />);

  wrapper.setState({
    nWrong: wrapper.props().maxWrong + 1
  });
  let img_elem = wrapper.find("img").first().html();
  expect(img_elem).toContain('src="6.jpg"');
});