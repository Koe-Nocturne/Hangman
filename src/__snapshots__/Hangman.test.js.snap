// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot after click 1`] = `
<div
  className="Hangman"
>
  <img
    src="1.jpg"
  />
  <p>
    Number wrong: 
    1
  </p>
  <p
    className="Hangman-word"
  >
    _
    _
    _
    _
    _
  </p>
  <p>
    <button
      disabled={false}
      key="a"
      onClick={[Function]}
      value="a"
    >
      a
    </button>
    <button
      disabled={false}
      key="b"
      onClick={[Function]}
      value="b"
    >
      b
    </button>
    <button
      disabled={false}
      key="c"
      onClick={[Function]}
      value="c"
    >
      c
    </button>
    <button
      disabled={false}
      key="d"
      onClick={[Function]}
      value="d"
    >
      d
    </button>
    <button
      disabled={false}
      key="e"
      onClick={[Function]}
      value="e"
    >
      e
    </button>
    <button
      disabled={false}
      key="f"
      onClick={[Function]}
      value="f"
    >
      f
    </button>
    <button
      disabled={false}
      key="g"
      onClick={[Function]}
      value="g"
    >
      g
    </button>
    <button
      disabled={false}
      key="h"
      onClick={[Function]}
      value="h"
    >
      h
    </button>
    <button
      disabled={false}
      key="i"
      onClick={[Function]}
      value="i"
    >
      i
    </button>
    <button
      disabled={false}
      key="j"
      onClick={[Function]}
      value="j"
    >
      j
    </button>
    <button
      disabled={false}
      key="k"
      onClick={[Function]}
      value="k"
    >
      k
    </button>
    <button
      disabled={false}
      key="l"
      onClick={[Function]}
      value="l"
    >
      l
    </button>
    <button
      disabled={false}
      key="m"
      onClick={[Function]}
      value="m"
    >
      m
    </button>
    <button
      disabled={false}
      key="n"
      onClick={[Function]}
      value="n"
    >
      n
    </button>
    <button
      disabled={false}
      key="o"
      onClick={[Function]}
      value="o"
    >
      o
    </button>
    <button
      disabled={false}
      key="p"
      onClick={[Function]}
      value="p"
    >
      p
    </button>
    <button
      disabled={false}
      key="q"
      onClick={[Function]}
      value="q"
    >
      q
    </button>
    <button
      disabled={false}
      key="r"
      onClick={[Function]}
      value="r"
    >
      r
    </button>
    <button
      disabled={false}
      key="s"
      onClick={[Function]}
      value="s"
    >
      s
    </button>
    <button
      disabled={false}
      key="t"
      onClick={[Function]}
      value="t"
    >
      t
    </button>
    <button
      disabled={false}
      key="u"
      onClick={[Function]}
      value="u"
    >
      u
    </button>
    <button
      disabled={false}
      key="v"
      onClick={[Function]}
      value="v"
    >
      v
    </button>
    <button
      disabled={false}
      key="w"
      onClick={[Function]}
      value="w"
    >
      w
    </button>
    <button
      disabled={false}
      key="x"
      onClick={[Function]}
      value="x"
    >
      x
    </button>
    <button
      disabled={false}
      key="y"
      onClick={[Function]}
      value="y"
    >
      y
    </button>
    <button
      disabled={true}
      key="z"
      onClick={[Function]}
      value="z"
    >
      z
    </button>
  </p>
</div>
`;
